packages/shorebird_code_push_client/lib/src/request_progress.dart

```diff
@@ -0,0 +1,26 @@
+/// A function that reports download or upload progress.
+typedef ProgressCallback = void Function(DataTransferProgress progress);
```

Comment:
Does this need to return a `DataTransferProgress` object or can it just be a `double`?

Reply:
It could be either. I thought it would be good to include the URL at least, as `progressStream` is per-client (not per-request), and this provides consumers a way to verify that the progress being reported is for the upload they care about (although this is not a problem we currently have), and once the URL was in there, it seemed like the other info was nice to have. 🤷